@xen-orchestra/web-core/lib/components/StackedBar.vue

```diff
@@ -0,0 +1,96 @@
+<!-- v1.0-->
+<template>
+  <div class="stacked-bar">
+    <div class="wrapper">
+      <div
+        v-for="(segment, index) in computedSegments"
+        :key="index"
+        :style="{ width: segment.percentage + '%' }"
+        class="segment typo c4-semi-bold"
+        :class="segment.color"
+      >
+        {{ segment.percentage.toFixed(0) }} %
+      </div>
+      <div
+        v-if="availableSegmentValue > 0"
+        :style="{ width: availableSegmentValue + '%' }"
+        class="segment available typo c4-semi-bold"
```

Comment:
As there is no text displayed, the typo classes are not useful:
```suggestion
        class="segment available"
```
Also this element is not useful if you set the default `background-color` on the component's root element.